### PR TITLE
Hook up Auto-materialize policy page to graphql

### PR DESCRIPTION
**Pull request by salazarm**
## Summary & Motivation

Hooking up the auto-materialize policy page to graphql. 

## How I Tested These Changes

Storybook
Screen Shot 2023-05-25 at 7 21 42 PM
Screen Shot 2023-05-25 at 7 21 35 PM

empty state:

Screen Shot 2023-05-25 at 7 20 28 PM

error states:

image

### CODE REVIEW

**Comment by salazarm** [comment]
Could use an improved empty state

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx [hunk 1]
```diff
@@ -24,92 +98,236 @@ export const AssetAutomaterializePolicyPage = () => {
...
-              <div
-                style={{borderRadius: '50%', height: '10px', width: '10px', color: Colors.Yellow50}}
-              />
-            )}
-          </CenterAlignedRow>
-        );
-      })}
+    <Box flex={{direction: 'column', grow: 1}} style={{overflowY: 'scroll'}}>
+      <Box
+        style={{flex: 1, minHeight: 0, overflowY: 'scroll'}}
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:154]: `overflowY: 'auto'` so that the scrollbar doesn't show up if it's not overflowing?

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx [hunk 2]
```diff
@@ -24,92 +98,236 @@ export const AssetAutomaterializePolicyPage = () => {
...
+        flex={{grow: 1, direction: 'column'}}
+      >
+        {evaluations.map((evaluation) => {
+          const date = dayjs.unix(evaluation.timestamp);
+          const formattedDate = date.format('lll');
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:159]: This might be inconsistent with other timestamp rendering throughout the app. The `<Timestamp>` or `<TimestampDisplay>` component (I should do something to disambiguate those...) would be more consistent.

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx [hunk 3]
```diff
@@ -24,92 +98,236 @@ export const AssetAutomaterializePolicyPage = () => {
...
+          const isSelected = selectedEvaluation === evaluation;
+          return (
+            <EvaluationRow
+              flex={{justifyContent: 'space-between'}}
+              key={evaluation.evaluationId + '' + Math.random()}
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:164]: Random key seems like it will guarantee re-renders you might not want...

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx [hunk 4]
```diff
@@ -24,92 +98,236 @@ export const AssetAutomaterializePolicyPage = () => {
...
+                <Icon name="auto_materialize_policy" size={24} />
+              ) : evaluation.numSkipped ? (
+                <div
+                  style={{
+                    borderRadius: '50%',
+                    height: '10px',
+                    width: '10px',
+                    color: Colors.Yellow500,
+                  }}
+                />
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:183]: Seems like we should have a dot component somewhere, but maybe not.

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx [hunk 5]
```diff
@@ -24,92 +98,236 @@ export const AssetAutomaterializePolicyPage = () => {
...
+                    width: '10px',
+                    color: Colors.Yellow50,
+                  }}
+                />
+              )}
+            </EvaluationRow>
+          );
+        })}
+      </Box>
+      {(queryResult.data?.autoMaterializeAssetEvaluations?.length || 0) > 0 ? (
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:198]: Should be enough just to check the truthiness of the `length` value 🤷

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx [hunk 6]
```diff
@@ -24,92 +98,236 @@ export const AssetAutomaterializePolicyPage = () => {
...
+      },
+    },
+  );
+  useQueryRefreshAtInterval(queryResult, FIFTEEN_SECONDS);
+  const {data, error} = queryResult;
+
+  React.useEffect(() => {
+    if (data?.assetNodeOrError.__typename === 'AssetNode') {
+      const max = data.assetNodeOrError.autoMaterializePolicy?.maxMaterializationsPerMinute;
+      if (max !== undefined && max !== null) {
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:228]: `if (typeof max === 'number') {`? Also, could `max` be zero, and if so, is that a problem?

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx [hunk 7]
```diff
@@ -24,92 +98,236 @@ export const AssetAutomaterializePolicyPage = () => {
...
-        />
-        <RightPanelDetail title="Maximum materializations per hour" value={2} tooltip="test" />
-      </RightPanelSection>
-      <RightPanelSection title="Freshness Policy">
-        <RightPanelDetail title="Maximum lag minutes" tooltip="test" value={2} />
-        <Box flex={{direction: 'column', gap: 8}}>
-          This asset will be considered late if it is not materialized within 2 minutes of it’s
-          upstream dependencies.
-          <ButtonLink>View upstream assets</ButtonLink>
+      {error ? (
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:242]: Just a personal thing, but I find the nested ternaries pretty hard to follow.

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx [hunk 8]
```diff
@@ -24,92 +98,236 @@ export const AssetAutomaterializePolicyPage = () => {
...
+                    >
+                      View documentation
+                    </ExternalAnchorButton>
+                  </div>
+                </Box>
+              }
+            />
+          )}
+          {data.assetNodeOrError.freshnessPolicy ? (
+            <RightPanelSection title="Freshness Policy">
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:294]: `Freshness policy`

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx [hunk 9]
```diff
@@ -24,92 +98,236 @@ export const AssetAutomaterializePolicyPage = () => {
...
+          <ErrorWrapper>{JSON.stringify(error)}</ErrorWrapper>
+        </Box>
+      ) : !data ? (
+        <Box flex={{direction: 'row', justifyContent: 'center'}} padding={{vertical: 24}}>
+          <Spinner purpose="section" />
         </Box>
-      </RightPanelSection>
+      ) : data.assetNodeOrError.__typename === 'AssetNotFoundError' ? (
+        <Redirect to="/assets" />
+      ) : (
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:252]: Worth showing an error message instead of a redirect?

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx [hunk 10]
```diff
@@ -155,43 +373,186 @@ const RightPanelDetail = ({
...
+          results.upstreamDataHasChangedSinceLatestMaterialization = true;
+          break;
+        case 'ParentOutdatedAutoMaterializeCondition':
+          results.waitingOnUpstreamData = true;
+          break;
+        case 'MaxMaterializationsExceededAutoMaterializeCondition':
+          results.exceedsXMaterializationsPerHour = true;
+          break;
+        default:
+          console.error('Unexpected condition', (cond as any).__typename);
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:434]: `assertUnreachable`?

File: js_modules/dagit/packages/core/src/assets/__stories__/AutoMaterializePolicyPage.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,72 @@
+import {MockedProvider} from '@apollo/client/testing';
+import {Box, ButtonGroup, Checkbox, JoinedButtons} from '@dagster-io/ui';
+import React from 'react';
+
+import {AutoMaterializePolicyType} from '../../graphql/types';
+import {AssetAutomaterializePolicyPage} from '../AutoMaterializePolicyPage/AssetAutomaterializePolicyPage';
+import {Evaluations, Policies} from '../__fixtures__/AutoMaterializePolicyPage.fixtures';
+
+// eslint-disable-next-line import/no-default-export
+export default {component: AssetAutomaterializePolicyPage};
```
Comment (hellendag) [inline js_modules/dagit/packages/core/src/assets/__stories__/AutoMaterializePolicyPage.stories.tsx:10]: Is this a new thing with Storybook 7?

File: js_modules/dagit/packages/ui/src/components/NonIdealState.tsx [hunk 12]
```diff
@@ -42,14 +42,16 @@ export const NonIdealState: React.FC<NonIdealStateProps> = ({
         <Spinner purpose="section" />
       ) : icon === 'no-results' ? (
         <Icon name="search" size={48} color={Colors.Gray400} />
-      ) : (
+      ) : icon ? (
         <Icon name={icon} size={48} color={Colors.Gray400} />
-      )}
+      ) : null}
```
Comment (hellendag) [inline js_modules/dagit/packages/ui/src/components/NonIdealState.tsx:47]: Why do we need a no-icon state?

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx (hunk 3 above)
Comment (salazarm) [reply js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:164]: Oops, I added the `Math.random()` while testing soemthing and forgot to remove it.

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx (hunk 9 above)
Comment (salazarm) [reply js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:252]: hmm, I suppose so

File: js_modules/dagit/packages/core/src/assets/__stories__/AutoMaterializePolicyPage.stories.tsx (hunk 11 above)
Comment (salazarm) [reply js_modules/dagit/packages/core/src/assets/__stories__/AutoMaterializePolicyPage.stories.tsx:10]: wdym?

File: js_modules/dagit/packages/ui/src/components/NonIdealState.tsx (hunk 12 above)
Comment (salazarm) [reply js_modules/dagit/packages/ui/src/components/NonIdealState.tsx:47]: The icon takes up too much space when shown in smaller spaces. In this case the right panel.

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx (hunk 6 above)
Comment (salazarm) [reply js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:228]: not a problem.

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx (hunk 2 above)
Comment (salazarm) [reply js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:159]: We use this format here: https://github.com/dagster-io/internal/blob/356395a1617046d164bb434982bfaf7f64a73089/dagster-cloud/js_modules/cloud-admin-panel/src/components/AccountReview.tsx#L35

File: python_modules/dagster-graphql/dagster_graphql/schema/auto_materialize_policy.py [hunk 13]
```diff
@@ -8,6 +8,7 @@
 
 class GrapheneAutoMaterializePolicy(graphene.ObjectType):
     policyType = graphene.NonNull(graphene.Enum.from_enum(AutoMaterializePolicyType))
+    maxMaterializationsPerMinute = graphene.Int()
```
Comment (salazarm) [inline python_modules/dagster-graphql/dagster_graphql/schema/auto_materialize_policy.py:11]: Should this be `NonNull` @johannkm

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx (hunk 10 above)
Comment (hellendag) [reply js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:430]: I think what we've found is that console.error does *not* get picked up 😢

File: js_modules/dagit/packages/core/src/assets/__stories__/AutoMaterializePolicyPage.stories.tsx (hunk 11 above)
Comment (hellendag) [reply js_modules/dagit/packages/core/src/assets/__stories__/AutoMaterializePolicyPage.stories.tsx:10]: Never mind, I see what I misunderstood

File: js_modules/dagit/packages/ui/src/components/NonIdealState.tsx (hunk 12 above)
Comment (hellendag) [reply js_modules/dagit/packages/ui/src/components/NonIdealState.tsx:47]: Hmm, maybe we need a smaller NonIdealState.

**Review by hellendag — APPROVED** [review]

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx [hunk 14]
```diff
@@ -24,92 +99,231 @@ export const AssetAutomaterializePolicyPage = () => {
...
-            ) : (
-              <div
-                style={{borderRadius: '50%', height: '10px', width: '10px', color: Colors.Yellow50}}
-              />
-            )}
-          </CenterAlignedRow>
-        );
-      })}
+    <Box flex={{direction: 'column', grow: 1}} style={{overflowY: 'auto'}}>
+      <Box style={{flex: 1, minHeight: 0, overflowY: 'auto'}} flex={{grow: 1, direction: 'column'}}>
```
Comment (bengotow) [inline js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:154]: I think this is correct but makes a good case for breaking out some sort of "Column" base component!

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx [hunk 15]
```diff
@@ -24,92 +99,231 @@ export const AssetAutomaterializePolicyPage = () => {
...
+              <Body style={{flex: 1}}>
+                {automaterializePolicyDescription(data.assetNodeOrError.autoMaterializePolicy)}
+              </Body>
+            </RightPanelSection>
+          ) : (
+            <NonIdealState
+              title="No Automaterialize policy found"
+              shrinkable
+              description={
+                <Box flex={{direction: 'column', gap: 8}}>
```
Comment (bengotow) [inline js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:270]: Is this also found on the asset Definition page still? If so maybe we should make these empty state components re-usable so we don't change the text in one place and forget it in the other.

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx [hunk 16]
```diff
@@ -24,92 +99,231 @@ export const AssetAutomaterializePolicyPage = () => {
...
+                  <div>
+                    An AutoMaterializePolicy specifies how Dagster should attempt to keep an asset
+                    up-to-date.
+                  </div>
+                  <div>
+                    <ExternalAnchorButton
+                      href="https://docs.dagster.io/_apidocs/assets#dagster.AutoMaterializePolicy"
```
Comment (bengotow) [inline js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:277]: Hmm it's a bit weird this is `_apidocs` -- i don't think that appears in other URLs we use, maybe this is private or in beta?

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx [hunk 17]
```diff
@@ -24,92 +99,231 @@ export const AssetAutomaterializePolicyPage = () => {
...
+}) => {
+  const queryResult = useQuery<GetPolicyInfoQuery, GetPolicyInfoQueryVariables>(
+    GET_POLICY_INFO_QUERY,
+    {
+      variables: {
+        assetKey,
+      },
+    },
+  );
+  useQueryRefreshAtInterval(queryResult, FIFTEEN_SECONDS);
```
Comment (bengotow) [inline js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:218]: I think this query is super lightweight so this is fine, but in most other places in the app we don't live reload the definition of the asset node, just the current state of it. (I think this is the only page where something like the automaterializePolicy autoreloads)

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx [hunk 18]
```diff
@@ -155,43 +369,186 @@ const RightPanelDetail = ({
...
+          break;
+        case 'ParentOutdatedAutoMaterializeCondition':
+          results.waitingOnUpstreamData = true;
+          break;
+        case 'MaxMaterializationsExceededAutoMaterializeCondition':
+          results.exceedsXMaterializationsPerHour = true;
+          break;
+        default:
+          console.error('Unexpected condition', (cond as any).__typename);
+          break;
```
Comment (bengotow) [inline js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:431]: This seems fine but a bit verbose, another way we might do this is say `requiredToMeetAFreshnessPolicy = conditions.some(c => c.__typename === 'FreshnessAutoMaterializeCondition')`

Doing it that way might be nice because each row component could take the evaluationData and the typename it's related to and do that check on it's own?

**Review by bengotow — APPROVED** [review]
This looks awesome! Sorry for the delay this morning, I left a few inline comments but nothing blocking or super major. Pretty awesome this connected so cleanly to the design you'd put together 👍

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx (hunk 16 above)
Comment (salazarm) [reply js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:277]: Hmm interesting, I found this page through google

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx (hunk 18 above)
Comment (salazarm) [reply js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:431]: I just get itchy when I think of all the extra loops we'd be doing but its nbd

File: js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx (hunk 15 above)
Comment (salazarm) [reply js_modules/dagit/packages/core/src/assets/AutoMaterializePolicyPage/AssetAutomaterializePolicyPage.tsx:270]: I'm not sure?
